crates/miden-objects/src/note/note_tag.rs

```diff
@@ -248,23 +276,54 @@ impl fmt::Display for NoteTag {
 
 impl From<u32> for NoteTag {
     fn from(value: u32) -> Self {
-        Self(value)
+        let prefix = value >> 30;
+        let lower_30_bits = value & 0x3fffffff; // Mask to get the lower 30 bits
+
+        match value & LOCAL_USECASE {
```

Comment:
```suggestion
        // Mask out the two most significant bits.
        match value & 0xC0000000 {
```

Nit: To me it's clearer if we inline this instead of using `LOCAL_USECASE`.